sdl_viewer/shaders/terrain.vs

```diff
@@ -0,0 +1,55 @@
+#version 410 core
+// The vertex positions are in grid coordinates, i.e. between 0 and grid_size (inclusive).
+layout(location = 0) in ivec3 aPos;
+
+// These are constant throughout the program.
+uniform double grid_size;
+uniform dvec3 terrain_origin_m;
+uniform double terrain_res_m;
+
+// The grid coordinates of the min corner of the currently visible terrain (the window).
+// Essentially, the terrain regular grid with resolution terrain_res_m and origin
+// terrain_origin_m, and the mesh is rendering a rectangle within that grid.
+uniform dvec2 terrain_pos;
+// These two are equal and set by the GlMovingWindowTexture.
+// They hold the coordinates (in pixels) of the min corner
+// of the texture within the sampler.
+uniform ivec2 height_texture_offset;
+uniform ivec2 color_texture_offset;
+// The full projective transform.
+uniform dmat4 world_to_gl;
```

Comment:
nit: I don't know about this repo, but in `avsoftware` we have the `x_from_y` convention for transforms: `gl_from_world`?

Reply:
I think that's nicer too, but wanted to keep it consistent with the rest of point_cloud_viewer, where it uses `y_to_x`.